modules/geo/src/internalClusterTest/java/org/opensearch/geo/search/aggregations/bucket/GeoTileGridIT.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geo.search.aggregations.bucket;
+
+import com.carrotsearch.hppc.ObjectIntHashMap;
+import com.carrotsearch.hppc.ObjectIntMap;
+import org.hamcrest.MatcherAssert;
+import org.opensearch.Version;
+import org.opensearch.action.index.IndexRequestBuilder;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.geo.GeoBoundingBox;
+import org.opensearch.common.geo.GeoPoint;
+import org.opensearch.common.geo.GeoShapeDocValue;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.geo.search.aggregations.bucket.geogrid.GeoGrid;
+import org.opensearch.geo.search.aggregations.bucket.geogrid.GeoGridAggregationBuilder;
+import org.opensearch.geo.search.aggregations.common.GeoBoundsHelper;
+import org.opensearch.geo.tests.common.AggregationBuilders;
+import org.opensearch.geo.tests.common.RandomGeoGenerator;
+import org.opensearch.geo.tests.common.RandomGeoGeometryGenerator;
+import org.opensearch.geometry.Geometry;
+import org.opensearch.geometry.Rectangle;
+import org.opensearch.search.aggregations.InternalAggregation;
+import org.opensearch.search.aggregations.bucket.GeoTileUtils;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.test.VersionUtils;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertSearchResponse;
+
+@OpenSearchIntegTestCase.SuiteScopeTestCase
+public class GeoTileGridIT extends AbstractBucketAggregationIntegTest {
```

Comment:
This duplicates logic w/ `GeoHashGridIT` so maybe we create a new Abstract class  (e.g.,  `AbstractGeoGridIntegTest.java`) that provides the common logic across the two? (e.g., `randomRectangle`, `prepareSingleValueGeoPointIndex`, `testGeoShapes`)?

Reply:
Some of the common logic has already been moved to an abstract class AbstractBucketAggregationIntegTest.class. As this was a geo module hence I thought to name it like this.

Now if we see other logic which is prepareSingleValueGeoPointIndex, testGeoShapes we can abstract some logic out but it will lead to multiple override functions and building tests in generic fashion.

I am not sure for integration tests I want to do that. The reason is if tests start to fail it becomes hard to debug currently we will get exact names that are failing. I understand there is some duplication but it provides some a better debugging experience. 

Having said that I will try to move more generic functions out to already created abstract class.

Function like randomRectangle will go to [RandomGeoGeometryGenerator](https://github.com/opensearch-project/OpenSearch/blob/56162a9f25b32a4f084bc21107e63aecc702ed4e/modules/geo/src/test/java/org/opensearch/geo/tests/common/RandomGeoGeometryGenerator.java#L39) class, which already have all helper functions for generating the random shapes.

Reply:
Let me abstract the logic for preparing the indexes for GeoShapes and GeoPoints. Other logic like creating rectangle will also be added.

Reply:
abstracted the code,